contrib/Shopify/sarama/sarama.go

```diff
@@ -44,6 +44,9 @@ func WrapPartitionConsumer(pc sarama.PartitionConsumer, opts ...Option) sarama.P
 				tracer.Tag("partition", msg.Partition),
 				tracer.Tag("offset", msg.Offset),
 			}
+			if cfg.analyticsRate > 0 {
```

Comment:
`0` is a valid analytics rate, we might need a way to distinguish between `0` and "not set".

Reply:
Having `0` or not setting the rate will have the same outcome. The agent will not pick the span in both cases.

Reply:
(unless of course it exists as a setting in `DD_APM_ANALYZED_SPANS` and the like)

Reply:
To follow up based on our chat: this is an edge-case where a user might want to override the "analyzed spans" setting made in the yaml file or by means of environment variable by setting 0 in the code, programmatically. Since the occurrence of this is quite minor, we will address the work as follow up.

Reply:
We can iterate in the future for that. This edge case should be supported, but it's not that relevant considering we're suggesting to send us all events.